src/server/scripts/Spells/spell_evoker.cpp

```diff
@@ -73,7 +78,64 @@ class spell_evo_glide : public SpellScript
     }
 };
 
+// 361469 - Living Flame (Red)
+class spell_evo_living_flame : public SpellScript
+{
+    PrepareSpellScript(spell_evo_living_flame);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo ({ SPELL_EVOKER_LIVING_FLAME_DAMAGE, SPELL_EVOKER_LIVING_FLAME_HEAL });
+    }
+
+    void HandleTarget(SpellEffIndex /*effIndex*/)
+    {
+        if (GetCaster()->IsFriendlyTo(GetHitUnit()))
+            GetCaster()->CastSpell(GetHitUnit(), SPELL_EVOKER_LIVING_FLAME_HEAL, true);
+        else
+            GetCaster()->CastSpell(GetHitUnit(), SPELL_EVOKER_LIVING_FLAME_DAMAGE, true);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_evo_living_flame::HandleTarget, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 361500 - Living Flame (damage)
+class spell_evo_living_flame_damage : public SpellScript
+{
+    PrepareSpellScript(spell_evo_living_flame_damage);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo ({ SPELL_EVOKER_ENERGIZING_FLAME, SPELL_EVOKER_LIVING_FLAME });
+    }
+
+    void HandleManaRestored(SpellEffIndex /*effIndex*/)
+    {
+        if (AuraEffect* auraEffect = GetCaster()->GetAuraEffect(SPELL_EVOKER_ENERGIZING_FLAME, EFFECT_0))
+        {
+            SpellInfo const* spellInfo = sSpellMgr->AssertSpellInfo(SPELL_EVOKER_LIVING_FLAME, GetCastDifficulty());
+
+            Optional<SpellPowerCost> cost = spellInfo->CalcPowerCost(POWER_MANA, false, GetCaster(), GetSpellInfo()->GetSchoolMask(), nullptr);
```

Comment:
(Not sure about what I’m about to say, maybe @Shauren or @mdX7 can correct me)

but isn’t it semantic wise wrong to have an optional here if it’s not really optional ?

Reply:
CalcPowerCost returns optional